### PR TITLE
(SLV-728) Update the scale summary script to include failing iterations with standard deviation

### PR DESCRIPTION
**Pull request by billclaytor**
The `util/report_utils/build_scale_csv_summary.rb` has been updated to include failing iterations with standard deviation for each column:
- replace 'average' with 'mean'
- calculate and include the standard deviation for each column
- include a table with the failing iteration

### CODE REVIEW

File: util/report_utils/build_scale_csv_summary.rb [hunk 1]
```diff
@@ -24,8 +20,134 @@
...
+# note: the template can be specified via the TEMPLATE_PATH environment variable
+#
+# @author Bill Claytor
+#
+# @param [String] parent_dir The parent directory containing the scale test result directories
+#
+# @return [void]
+#
+# @example
+#   build_report(parent_dir)
```
Comment (johnduarte) [inline util/report_utils/build_scale_csv_summary.rb]: does not match method name.

File: util/report_utils/build_scale_csv_summary.rb [hunk 2]
```diff
@@ -24,8 +20,134 @@
...
+# @author Bill Claytor
+#
+# @param [String] csv_path The scale test results CSV file path
+#
+# @return [String] The CSV data converted to an HTML table
+#
+# @example
+#    table = process_results_csv(csv_path)
+#
+def process_results_csv(csv_path)
```
Comment (johnduarte) [inline util/report_utils/build_scale_csv_summary.rb]: nitpick: You have one `process_*` method that has a return value (process_result_csv) and one that has no return value (process_result_dirs).  This UX is inconsistent.  Consider altering the naming convention for these methods.

File: util/report_utils/build_scale_csv_summary.rb (hunk 1 above)
Comment (billclaytor) [reply util/report_utils/build_scale_csv_summary.rb]: Updated

File: util/report_utils/build_scale_csv_summary.rb (hunk 2 above)
Comment (billclaytor) [reply util/report_utils/build_scale_csv_summary.rb]: Renamed `process_result_csv` to `extract_table_from_results_csv`.

File: util/report_utils/build_scale_csv_summary.rb [hunk 3]
```diff
@@ -24,8 +20,134 @@
...
+# @return [void]
+#
+# @example
+#   build_report(parent_dir)
+#
+def build_scale_csv_summary(parent_dir)
+  puts "building scale test csv summary report for parent dir: #{parent_dir}"
+  puts
+
+  parent_name = File.basename(parent_dir)
```
Comment (billclaytor) [inline util/report_utils/build_scale_csv_summary.rb]: This can result in awkward report names; updated to use a generic name.

**Pull request merged**